tests/testParser.hs

```diff
@@ -1,11 +1,12 @@
 {-# LANGUAGE OverloadedStrings #-}
-
+{-# LANGUAGE TypeApplications  #-}
```

Comment:
We can run the tests by adding explicit type applications. However, while testParser.hs still works, pretty much all the tests in test.hs failed after refactoring. Also, it would be nice if we could get rid of the unnecessary annotations. One way is to make all parsing functions monomorphic, which kind of makes sense since the parser for liquid-fixpoint is never invoked on a particular source language.